lib/readline.js

```diff
@@ -47,9 +47,7 @@ function Interface(input, output, completer, terminal) {
   }
   historySize = historySize || kHistorySize;
 
-  completer = completer || function() { return []; };
-
-  if (typeof completer !== 'function') {
+  if (completer && typeof completer !== 'function') {
```

Comment:
Let's wait for others to take a look at this. I feel that we should allow only if it is a function.